command/server_util.go

```diff
@@ -14,63 +13,80 @@ var (
 	onEnterprise = false
 )
 
-func adjustCoreForSealMigration(ctx context.Context, core *vault.Core, coreConfig *vault.CoreConfig, seal vault.Seal, config *server.Config) error {
+func adjustCoreForSealMigration(ctx context.Context, core *vault.Core, barrierSeal, unwrapSeal vault.Seal) error {
 	existBarrierSealConfig, existRecoverySealConfig, err := core.PhysicalSealConfigs(context.Background())
 	if err != nil {
 		return fmt.Errorf("Error checking for existing seal: %s", err)
 	}
-	var existSeal vault.Seal
-	var newSeal vault.Seal
-	if existBarrierSealConfig != nil && existBarrierSealConfig.Type != vaultseal.HSMAutoDeprecated &&
-		(existBarrierSealConfig.Type != seal.BarrierType() ||
-			config.Seal != nil && config.Seal.Disabled) {
-		// If the existing seal is not Shamir, we're going to Shamir, which
-		// means we require them setting "disabled" to true in their
-		// configuration as a sanity check.
-		if (config.Seal == nil || !config.Seal.Disabled) && existBarrierSealConfig.Type != vaultseal.Shamir {
-			return errors.New(`Seal migration requires specifying "disabled" as "true" in the "seal" block of Vault's configuration file"`)
-		}
 
-		// Conversely, if they are going from Shamir to auto, we want to
-		// ensure disabled is *not* set
-		if existBarrierSealConfig.Type == vaultseal.Shamir && config.Seal != nil && config.Seal.Disabled {
-			coreConfig.Logger.Warn(`when not migrating, Vault's config should not specify "disabled" as "true" in the "seal" block of Vault's configuration file`)
+	// If we don't have an existing config or if it's the deprecated auto seal
+	// which needs an upgrade, skip out
+	if existBarrierSealConfig == nil || existBarrierSealConfig.Type == vaultseal.HSMAutoDeprecated {
+		return nil
+	}
+
+	if unwrapSeal == nil {
+		// We have the same barrier type and the unwrap seal is nil so we're not
+		// migrating from same to same, IOW we assume it's not a migration
+		if existBarrierSealConfig.Type == barrierSeal.BarrierType() {
 			return nil
 		}
 
-		if existBarrierSealConfig.Type != vaultseal.Shamir && existRecoverySealConfig == nil {
-			return errors.New(`Recovery seal configuration not found for existing seal`)
+		// If we're not coming from Shamir, and the existing type doesn't match
+		// the barrier type, we need both the migration seal and the new seal
+		if existBarrierSealConfig.Type != vaultseal.Shamir && barrierSeal.BarrierType() != vaultseal.Shamir {
+			return errors.New(`Trying to migrate from auto-seal to auto-seal but no "disabled" seal stanza found`)
```

Comment:
I see that the other pre-existing errors in this file are capitalized, so this is consistent. Would now be a good time to partially or fully convert the errors in this file to lowercase to be more Go-y?

Reply:
This doesn't go to the logs and doesn't go over the API, this goes directly to the UI/CLI on launch. Those errors we use proper capitalization.

Reply:
Ah! I didn't realize that! :notes: The more you know.....